src/vcpkg/base/system.process.cpp

```diff
@@ -476,18 +476,24 @@ namespace vcpkg
         }
 
         std::atomic<size_t> work_item{0};
+        std::mutex mtx;
         const auto num_threads =
             static_cast<size_t>(std::max(1, std::min(get_concurrency(), static_cast<int>(cmd_lines.size()))));
 
+        vcpkg::Checks::check_exit(VCPKG_LINE_INFO, num_threads > 1, "num_threads is %d", num_threads);
```

Comment:
This is weird: `num_threads` should be an `unsigned int`. Negative number of threads doesn't make any sense. Interestingly, `get_concurrency()` also returns `int` instead which doesn't make any sense either. If `-1` actually is a valid state, I would return an `Optional<unsigned int>` instead.